opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/ListTaskPresenter.java

```diff
@@ -480,6 +502,19 @@ public int getInterventionLabel() {
         return org.smartregister.reveal.util.Utils.getInterventionLabel();
     }
 
+    @Override
+    public void onMarkStructureInactiveConfirmed() {
+        listTaskInteractor.markStructureAsInactive(selectedFeature);
+
+    }
+
+    @Override
+    public void onStructureMarkedInactive() {
+        selectedFeature.addStringProperty(STATUS, INACTIVE);
+        selectedFeature.addStringProperty(BUSINESS_STATUS, INACTIVE);
+        refreshStructures(true);
```

Comment:
could you refresh only the 1 amended structure instead of querying the database again, while its possible to know the structure task details